### PR TITLE
fixup! There is a new server management module placed in the plugins …

### PR DESCRIPTION
**Pull request by rjeffman**
Please, add this patch to add the missing variables to automoutmap module.

This patch contains a `fixup` to the commit for the module, so, if applied, squash the commit of this PR into the one on the original branch.

### CODE REVIEW

File: README-automountmap.md [hunk 1]
```diff
@@ -86,9 +103,18 @@ Variable | Description | Required
...
+`map_type` | Type of automount map. Valid values are `direct` and `indirect`. Defaults to `direct`. Type `indirect` can only be used if automount map does not exist.| no
+`parentmap` | Name of parent automount map. Defaults to `auto.master`. Only valid if type is `indirect`. | no
+`mount` \| `key` | Mount point. Only valid if type is `indirect`. | yes (if type is `indirect`)
 `state` | The state to ensure. It can be one of `present`, or `absent`, default: `present`. | no
 
 
+Notes
+=====
+
+The use of `map_type: indirect` is not idempotent. If running the same playbook with `map_type: indirect` and `state: present`, assuming the `mapname` does not exist prior to the first execution, the first run will return `changed: True`, and the second one will return an error.
```
Comment (chr15p) [inline README-automountmap.md]: Returning an error in the second run is Not Good. If the api can't be used indempotently then can we code around it in the module to detect if the map is already there and not change it?

File: tests/automount/test_automountmap.yml [hunk 2]
```diff
@@ -48,9 +58,9 @@
       location: TestLocation
       desc: "this is a changed description that should be deleted by the test"
     register: result
-    failed_when: result.failed or  not result.changed
+    failed_when: result.failed or not result.changed
 
-  - name: ensure map TestMap has a different description
+  - name: ensure map TestMap has a different description, again
```
Comment (chr15p) [inline tests/automount/test_automountmap.yml]: its not really ensuring it again,  should be more like "ensure map TestMap description has been changed"

File: tests/automount/test_automountmap.yml [hunk 3]
```diff
@@ -133,3 +214,14 @@
     register: result
     failed_when: result.failed or result.changed
 
+  # CLEAN UP
+
+  - name: ensure test maps are absent
```
Comment (chr15p) [inline tests/automount/test_automountmap.yml]: The clean up is done by removing the location (sorry "ensuring its absent")  in the 2 tasks above this which remove the location and clean up all the maps contained in it. So this task should probably be removed or (better) moved up above the "ensure location TestLocation is absent" task

**Review by chr15p — CHANGES_REQUESTED** [review]

File: tests/automount/test_automountmap.yml (hunk 3 above)
Comment (rjeffman) [reply tests/automount/test_automountmap.yml]: This duplication "won't hurt", and there is ongoing work to force cleanup code for the tests if it fails.

File: tests/automount/test_automountmap.yml (hunk 2 above)
Comment (rjeffman) [reply tests/automount/test_automountmap.yml]: The "again" is because we are re-running the same task, and we use the same in other tests. Feel free to improve the message.

File: README-automountmap.md (hunk 1 above)
Comment (rjeffman) [reply README-automountmap.md]: I agree, but the information added by `automountmap-add-indirect` is not found on the map itself, but on its parent (given by `parentmap` or, by default, `auto.master`.

The alternative seems to be seaching for all keys, in all maps, to find the proper one, but this is error prone, as different maps can have the same keys, and you wouldn't know which map to modify.

This is unfortunate that the operation is not idempotent, but it is not the only one we currently have.

Comment (chr15p) [reply README-automountmap.md]: description is indempotent so we can change that easily. For the keys it creates I think I prefer just silently ignoring any changes in them rather than breaking the playbook, they can be sorted out with an automountkey task (*cough* when its merged *cough*)  rather than explicitly failing the task and thereby failing the whole playbook. 

an alternative might be to check for the existence of the named key (we have that info) and if it doesn't exist create it, ignoring any previous keys we might have created previously. It leaves some garbage keys laying around (the one we meant to modify) but tbh if you change the location you are going to get garbage maps left anyway because we cant tidy those up either. 

indirect maps are a real bodge in the ldap schema aren't they :( (/me gives a hard stare to whoever wrote rfc2307 )

Comment (rjeffman) [reply README-automountmap.md]: I don't like having orphaned keys around when we are talking about configuration automation.

As we want the automount patches to be merged, I suggest implementing support only for `map_type: direct` and design the `indirect` part later.

Do you agree?

Comment (chr15p) [reply README-automountmap.md]: makes sense.  

The mapname a key was created with seems to be stored in the automountInformation field, so we might be able to associated the map and key back together and make changes, but it needs some testing to find all the edge cases and at the moment I have no time to work on it :(

so lets do the direct maps and let someone do the indirect ones at a later date
